sklearn/tests/test_isotonic.py

```diff
@@ -676,3 +676,22 @@ def test_isotonic_2darray_more_than_1_feature():
 
     with pytest.raises(ValueError, match=msg):
         iso_reg.transform(X_2d)
+
+
+def test_isotonic_regression_sample_weight_not_overwritten():
+    """Check that calling fitting function of isotonic regression will not
+    overwrite `sample_weight`.
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/20508
+    """
+    from sklearn.datasets import make_regression
+
+    X, y = make_regression(n_samples=10, n_features=1, random_state=41)
+    sample_weight_original = np.ones(shape=y.shape)
```

Comment:
I would not set it to only ones because it likely that this specific vector would remain unchanged. Maybe change 1 value or make a vector of random values